monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js

```diff
@@ -87,7 +89,26 @@ function deleteAbstractAgentEventFields(myObj) {
   return tempObj;
 }
 
-const renderEventSpecificFields = (val) => filterEventSpecificFields(val);
+function redactSecretsInEventFields(myObj) {
+  let tempObj = JSON.parse(JSON.stringify(myObj)); /* deepcopy */
+
+  let stolenCredentialsFieldName = "stolen_credentials";
+  let secretFieldName = "secret";
+
+  if (tempObj.hasOwnProperty(stolenCredentialsFieldName)) {
+    for (let stolenCredential of tempObj[stolenCredentialsFieldName]) {
+      let secrets = stolenCredential[secretFieldName];
+      for (let secretType in secrets) {
+        let redactedSecret = "*".repeat(stolenCredential[secretFieldName][secretType].length);
```

Comment:
Can we have some fixed length as this gives away the actual length of the credential?

Reply:
I don't think it really matters.